local-repo-dev-sync.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+# SPDX-License-Identifier: MIT
+
+set -euo pipefail
+
+if [ "${DEBUG:-false}" = true ]; then
+    set -x
+fi
+
+LSR_GH_ORG=${LSR_GH_ORG:-linux-system-roles}
+LSR_BASE_DIR=${LSR_BASE_DIR:-~/linux-system-roles}
+
+requiredcmds="hub jq"
+missingcmds=""
+
+for cmd in $requiredcmds; do
+    if ! type -p "$cmd" > /dev/null 2>&1; then
+        missingcmds="$missingcmds $cmd"
+    fi
+done
+
+if [ -n "$missingcmds" ]; then
+    echo ERROR: this script requires the following commands: "$missingcmds"
+    echo e.g. on Fedora - dnf -y install "$missingcmds"
+    exit 1
+fi
+
+if [ ! -d "$LSR_BASE_DIR" ]; then
+    mkdir -p "$LSR_BASE_DIR"
+fi
+
+##
+# GET (list) a given github url, using the page interface to get all
+# of the pages, and using a given jq filter to extract fields
+#
+#    $1 - github url e.g. orgs/linux-system-roles/repos
+#    $2 - jq filter e.g. '.[].name'
+#
+# echos to stdout the filtered results.
+# NOTE: the -e argument to jq tells jq to exit with an error code
+# if the input is empty - this breaks the loop
+gh_get_all() {
+    local uri="$1"
+    local filter="$2"
+    local page=1
+    while hub api "${uri}?page=$page" | jq -e -r "$filter"; do
```

Comment:
For this to work, I needed to create a Personal access token in GitHub and set the `GITHUB_TOKEN` environment variable in my OS to point to the token. Without these steps, the script hangs at this line without any error message.

Reply:
When I've set the GITHUB_TOKEN var, the script worked fine.

Reply:
hmm - also, hub uses `~/.config/hub`:
```
github.com:
- user: richm
  oauth_token: xxxxxxxxxxxxxx
  protocol: https
```
I thought there was some hub command like hub login to create the initial token and write the file - It's been a while since I did the hub setup - maybe on the hub man page it specifies this

Reply:
from the `hub` man page:
```
   GitHub OAuth authentication
       Hub will prompt for GitHub username  password the first time it needs
       to access the API and exchange it for an OAuth token, which it saves in
       ~/.config/hub.
```
So I guess just run something like `hub api /repos/linux-system-roles/mssql` and it should prompt for username and password, and create `~/.config/hub` for you

Reply:
When I provide my credentials to `hub api`, it return an error saying `Not Found`:
```
$ hub api "orgs/linux-system-roles/repos?page=1" 
github.com username: spetrosi@redhat.com
github.com password for spetrosi@redhat.com (never stored): 
Not Found
```
I guess that this might happen because I have two-factor auth configured in GitHub, can this be the case? I only managed to configure this by creating a token and setting the env var.

Reply:
> When I provide my credentials to hub api, it return an error saying Not Found:

I suppose it could be related to 2FA, but I would expect a 401 or 403 instead of Not Found - you could use `hub api -i ...` and see if you can determine what is the actual error from the headers returned.

But at any rate - if you are ok with setting the token environment variable every time, or if you create a `~/.config/hub` with your token

Reply:
I am completely ok with doing this, but the script must say that there is such a requirement. Otherwise, there is a gap because now the script only asks to install hub.